Ćwiczenie 1: Series — tworzenie i operacje (20 min)

In [1]:
import pandas as pd
import numpy as np



In [3]:
# Series z listy — domyślny indeks (0, 1, 2, ...)
oceny = pd.Series([4.5, 3.0, 5.0, 3.5, 4.0])
print(f'{oceny}')

0    4.5
1    3.0
2    5.0
3    3.5
4    4.0
dtype: float64


Zadanie 1a: Utwórz Series pensje z dict:

In [15]:
# Klucze: 'Anna', 'Jan', 'Ewa', 'Marek', 'Kasia'
# Wartości: 5500, 7200, 4800, 9100, 6300

temp = {'Anna': 5500,
        'Jan':7200,
        'Ewa':4800,
        'Marek':9100,
        'Kasia':6300
    }
pensje = pd.Series(data= temp, index = ['Anna', 'Jan', 'Ewa', 'Marek', 'Kasia'])
print(pensje.to_string())

Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300


#Zadanie 1b: Odpowiedz na pytania o pensje :

In [14]:
# 1. Wyświetl imię osoby zarabiającej najwięcej (użyj idxmax())
print(pensje.idxmax())
# 2. Oblicz średnią pensję
print(pensje.mean())
# 3. Wyświetl imiona osób zarabiających powyżej średniej (filtrowanie boolean)
maska = pensje[pensje > 6580]
print(maska.to_string())
# 4. Policz ile osób zarabia poniżej 6000 (użyj len() lub .sum())
maska1 = pensje < 6000
print(maska1.sum())

Marek
6580.0
Jan      7200
Marek    9100
2


Zadanie 1c: Oblicz pensje po zmianach:

In [36]:
# 1. Oblicz pensje po podwyżce 10% dla wszystkich (pensje * 1.10)
new_pensja= pensje* 1.10
print(f'Nowa pensja wynosi\n{new_pensja.to_string()}')
# 2. Oblicz pensje po premii 500 zł dla wszystkich (pensje + 500)
prem = pensje + 500
print(f'\nPensje po premii wynoszą\n{prem.to_string()}')
# 3. Oblicz nową pensję = (pensja * 1.10) + 500
temp2 = (pensje * 1.10) + 500
print(f'\nNowe pensje wynoszą\n{temp2.to_string()}')
# 4. Wyświetl ranking nowych pensji od najwyższej (użyj sort_values)
sort= temp2.sort_values(ascending=False)
print(f'\n{sort.to_string()}')

Nowa pensja wynosi
Anna      6050.0
Jan       7920.0
Ewa       5280.0
Marek    10010.0
Kasia     6930.0

Pensje po premii wynoszą
Anna     6000
Jan      7700
Ewa      5300
Marek    9600
Kasia    6800

Nowe pensje wynoszą
Anna      6550.0
Jan       8420.0
Ewa       5780.0
Marek    10510.0
Kasia     7430.0

Marek    10510.0
Jan       8420.0
Kasia     7430.0
Anna      6550.0
Ewa       5780.0


Ćwiczenie 2: DataFrame — tworzenie i atrybuty (20 min)

In [ ]:
# Dane o produktach sklepu internetowego
sklep = pd.DataFrame({
'produkt': ['Laptop', 'Tablet', 'Smartfon', 'Słuchawki', 'Monitor'],
'cena': [3500, 1800, 2500, 350, 2200],
'stan_magazynowy': [45, 120, 200, 500, 75],
'kategoria': ['Komputery', 'Mobilne', 'Mobilne', 'Akcesoria', 'Komputery']
})


Zadanie 2a: Zbadaj DataFrame sklep :

In [47]:
# 1. Wyświetl kształt tabeli (shape)
print(sklep.shape)
# 2. Wyświetl nazwy kolumn (columns)
print(sklep.columns)
# 3. Wyświetl typy danych w każdej kolumnie (dtypes)
print(sklep.dtypes)
# 4. Wyświetl pierwsze 3 wiersze (head)
print(sklep.head(3))


(5, 4)
Index(['produkt', 'cena', 'stan_magazynowy', 'kategoria'], dtype='str')
produkt              str
cena               int64
stan_magazynowy    int64
kategoria            str
dtype: object
    produkt  cena  stan_magazynowy  kategoria
0    Laptop  3500               45  Komputery
1    Tablet  1800              120    Mobilne
2  Smartfon  2500              200    Mobilne


Zadanie 2b: Dodaj nowe kolumny do DataFrame sklep :

In [52]:
# 1. Dodaj kolumnę 'wartosc_magazynu' = cena * stan_magazynowy
df = sklep
df['wartosc_magazynu'] = df['cena'] * df['stan_magazynowy']

# 2. Dodaj kolumnę 'cena_z_vat' = cena * 1.23
df['cena_z_vat'] = df['cena'] * 1.23
# 3. Wyświetl cały DataFrame z nowymi kolumnami
print(sklep)


     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0


Krok 2 — Wczytanie CSV


In [53]:
# Wczytaj dataset tips
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url)


Zadanie 2c: Podstawowa eksploracja datasetu tips

In [59]:
# 1. Wyświetl ile wierszy i kolumn ma dataset (shape)
tips.shape
# 2. Wyświetl ostatnie 5 wierszy (tail)
tips.tail(5)
# 3. Wyświetl 3 losowe wiersze (sample)
tips.sample(3)
# 4. Wyświetl typy danych w każdej kolumnie (dtypes)
tips.dtypes


total_bill    float64
tip           float64
sex               str
smoker            str
day               str
time              str
size            int64
dtype: object

Ćwiczenie 3: Eksploracja datasetu — samodzielna analiza (30 min)

In [6]:
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)


Zadanie 3a: Pierwszy rzut oka


In [7]:
# 1. Wyświetl head(10)
print(penguins.head(10))
# 2. Wyświetl shape
print(penguins.shape)
# 3. Uruchom info() — ile brakujących wartości widzisz?
print(penguins.info())

  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       36

Zadanie 3b: Statystyki opisowe

In [10]:
df= penguins
# 1. Uruchom describe() — co mówią ci średnie i odchylenia?
print(penguins.describe())
# 2. Uruchom describe(include='all') — co dodają kolumny tekstowe?
print(f'\n{penguins.describe(include='all')}')
# 3. Jaka jest mediana masy ciała (body_mass_g)?
print(f'\n{df['body_mass_g'].median()}')

print(df['body_mass_g'].mean())

       bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

       species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count      344     344      342.000000     342.000000         342.000000   
unique       3       3             NaN            NaN                NaN   
top     Adelie  Biscoe             NaN            NaN                NaN   
freq       152     168             NaN            NaN                NaN  

Zadanie 3c: Rozkłady wartości

In [82]:
# 1. Ile jest pingwinów każdego gatunku? (value_counts na 'species')
print(f'{penguins.value_counts('species').to_string()}')

# 2. Ile pingwinów na każdej wyspie? (value_counts na 'island')
print(f'\n {penguins.value_counts('island').to_string()}')

# 3. Rozkład płci (value_counts na 'sex') — czy jest zbalansowany?
print(f'\n {penguins.value_counts('sex').to_string()}')

# 4. Jaki procent to gatunek Adelie? (normalize=True)
print(f'\n {penguins.value_counts('species', normalize=True).to_string()}')


species
Adelie       152
Gentoo       124
Chinstrap     68

 island
Biscoe       168
Dream        124
Torgersen     52

 sex
MALE      168
FEMALE    165

 species
Adelie       0.441860
Gentoo       0.360465
Chinstrap    0.197674


Zadanie 3d: Brakujące dane


In [85]:
# 1. Policz brakujące wartości w każdej kolumnie (isna().sum())
print(f'{penguins.isna().sum().to_string()}')
# 2. Jaki procent danych brakuje w kolumnie 'sex'?
print(f'\n {penguins.isna().value_counts('sex', normalize=True).to_string()}')
# Podpowiedź: braki / len(penguins) * 100


species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11

 sex
False    0.968023
True     0.031977


Zadanie 3e: Wnioski (refleksja) W komórce Markdown napisz minimum 5 obserwacji o datasecie. Każda obserwacja to jedno zdanie poparte liczbą z twoich obliczeń
powyżej. Przykłady:

Dataset zawiera 340 pingwinów z 3 gatunków..."
"Najczęstszy gatunek to Adelie (152 osobników, 44%)"
"Brakuje danych w kolumnachbill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex  (łącznie 19 braków)"
"Średnia masa ciała to 4201 g (mediana: 4050 g)"
"Pingwiny mieszkają na 3 wyspach, najliczniej na Biscoe"
